### ISSUE TITLE
Slider is broken when unit of measurement is "%"

### ISSUE THREAD

**Issue by lmagyar** [nielsfaber/scheduler-card#401]
Scheduler v3.1.1
Card v2.1.2 (browser cache deleted)
HA 2021.5.3 OS

Test config:
```yaml
input_number:
  test:
    min: 40
    max: 80
    name: Test
    unit_of_measurement: "%"
```

How to replicate:
- add schedule, select "Set value" action, use the slider

Bug:
- slider can move 0 to 100% (not 40 to 80%)
- values out of min/max range are stored in the schedule, but displayed only a value within the range
  - with above example, you can set a value of 23%, it is stored in the schedule, though only "40%" is displayed everywhere

**Comment by nielsfaber** [comment]
Actually it is not really a bug, since I deliberately did this (my bad!).
The reason is that e.g. light brightness is defined from 0-255 in HA.
By assigning `unit: '%'` the card will normalize/scale the range to 0-100 (which IMHO is more intuitive).

I understand that it might not be particularly desirable for your case, but currently I don't know how to combine both worlds.
Possibly I could define some wildcard unit `__PERCENT__` for the light case and solve it internally.
What do you think is best?

**Comment by lmagyar** [comment]
I think the bug I described is not exactly a simple 0-100 -> 0-255 conversion, the slider was able to move out of the min-max ranges, a value outside of the min-max range was possible to set and store, etc.

But if it is caused by the conversion, my 2 cents are below:
- When unit is not defined, the slider moves between min and max values, it's fine.
- When unit is °C, °F, x, foo, bar, etc. the slider behaves the same, moves between min and max values, the unit is displayed after the value as a plain string.
- I think when the unit is %, it should behave the same.
- If an entity requires special handling, some other attribute of it should turn on the special handling, not the unit of measurement.

But... Nobody knows how many people out there rely on the current behavior. So I think that ship has sailed, we need some workaround.

The algorithm could be to **_"turn on the special handling if the unit is %, except when ??????"_** The problem, I can say now "except when the entity's domain is humidifier". But what about the above input_number if somebody uses it for eg. some default humidity value setting? Maybe **_"except when device_class is humidity, humidifier, dehumidifier"_**? People can customize even an input_number if they need this "normal" behavior. But it feels hacky.

**Comment by lmagyar** [comment]
I'm testing it, and you are right, it has a conversion problem also! :) The humidifier is set to a strange value, the slider says eg. 45%, but the set value is 58%, so it sets it to "min + scheduled_value * (max - min)", it is definitively useful for a lamp, but totally mixes up a humidifier. :/

**Comment by nielsfaber** [comment]
I decided to change this behaviour since I agree it is not OK at the moment.
The card will not longer look for percentage unit.
An option `scale_factor` is added (which is set at 2.55 for `light` brightness), to define scaling between what is displayed and what is used in `service_data`.
I think it is much more transparent that way.

**Comment by lmagyar** [comment]
It is fixed in v2.1.4.
